### PR TITLE
Emit interceptor info correctly when invocation expr is on separate line

### PR DESCRIPTION
**Pull request by layomia**
Addresses https://github.com/dotnet/runtime/issues/90851. RC2 candidate.

### CODE REVIEW

File: src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/ConfigurationBinderTests.Generator.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        [Fact]
+        public void GeneratorHandlesInvocationsOnNewline()
+        {
+            IConfiguration configuration = TestHelpers.GetConfigurationFromJsonString(@"{""Longitude"":1,""Latitude"":2}");
+
+            GeolocationRecord record = configuration.Get<GeolocationRecord>();
+            Verify();
+
+            record = (GeolocationRecord)configuration
+                .Get(typeof(GeolocationRecord), _ => { });
```
Comment (ericstj) [inline src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/ConfigurationBinderTests.Generator.cs:20]: Should we be testing other permutations of line breaks that are permitted by the language to make sure we handle all of them?

Comment (tarekgh) [reply src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/ConfigurationBinderTests.Generator.cs:20]: +1 

```C#
record = (GeolocationRecord)configuration.
                Get(typeof(GeolocationRecord), _ => { });

record = (GeolocationRecord)
                configuration.
                Get(typeof(GeolocationRecord), _ => { });

record = (GeolocationRecord)
                configuration
                .Get(typeof(GeolocationRecord), _ => { });
```

**Comment by tarekgh** [comment]
Looks this change broke the binding to the static properties. 

https://[helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-91107-merge-2933836994b843458b/Microsoft.Extensions.Configuration.Binder.SourceGeneration.Tests/1/console.7dc38e07.log?helixlogtype=result](https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-91107-merge-2933836994b843458b/Microsoft.Extensions.Configuration.Binder.SourceGeneration.Tests/1/console.7dc38e07.log?helixlogtype=result)
```
    Microsoft.Extensions.SourceGeneration.Configuration.Binder.Tests.ConfigurationBinderTests.GetCanReadStaticProperty [FAIL]
      Assert.Equal() Failure
      Expected: stuff
      Actual:   (null)
      Stack Trace:
        /_/src/libraries/Microsoft.Extensions.Configuration.Binder/tests/Common/ConfigurationBinderTests.cs(751,0): at Microsoft.Extensions.SourceGeneration.Configuration.Binder.Tests.ConfigurationBinderTests.GetCanReadStaticProperty()
        /_/src/mono/System.Private.CoreLib/src/System/Reflection/MethodBaseInvoker.Mono.cs(22,0): at System.Reflection.MethodBaseInvoker.InterpretedInvoke_Method(Object obj, IntPtr* args)
        /_/src/libraries/System.Private.CoreLib/src/System/Reflection/MethodBaseInvoker.cs(57,0): at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
    Microsoft.Extensions.SourceGeneration.Configuration.Binder.Tests.ConfigurationBinderCollectionTests.DoesNotBindUninstantiatedISetWithUnsupportedKeys [SKIP]
```

```C#
        [Fact]
        public void GetCanReadStaticProperty()
        {
            var dic = new Dictionary<string, string>
            {
                {"StaticProperty", "stuff"},
            };
            var configurationBuilder = new ConfigurationBuilder();
            configurationBuilder.AddInMemoryCollection(dic);
            var config = configurationBuilder.Build();
            var options = new ComplexOptions();
            config.Bind(options);

            Assert.Equal("stuff", ComplexOptions.StaticProperty);
        }
```

**Review by tarekgh — APPROVED** [review]

**Comment by layomia** [comment]
@tarekgh hmm don't think so. We maintain the same binding behavior. Taking a look; might file an issue.

**Comment by layomia** [comment]
@tarekgh mono issues don't surface after rebasing on top of main.

Test failure appears related to https://github.com/dotnet/runtime/issues/87783.

**Comment by layomia** [comment]
Needs backport to RC2 to consider fixed for 8.0.

**Pull request merged**